### ISSUE TITLE
CMake issue with Eigen3

### ISSUE THREAD

**Issue by tvatter** [PatWie/CppNumericalSolvers#68]
Hi,

At line 33 of CMakeLists.txt, I think that

link_libraries(::Eigen3)

should be replaced by

include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})

Since Eigen3 is header only, I do not get the linking through cmake's link_libraries, which gives returns an error on my computer (OSX).

Best,
Thibault

**Comment by PatWie** [comment]
@tvatter  Is this still the case in
https://github.com/PatWie/CppNumericalSolvers/blob/master/CMakeLists.txt#L33
?

**Comment by tvatter** [comment]
Yes, it is.

**Comment by jdumas** [comment]
Note that with recent versions of Eigen (3.2.9 and 3.3 I think), Eigen provides a `Eigen3Config.cmake` which installs as parts of Eigen, and this file defines an interface target for Eigen. Which means we can now write:

```
find_package(Eigen3 REQUIRED)
target_link_libraries(${PROJECT_NAME} PUBLIC Eigen3::Eigen)
```

It will pull any dependency it can find (e.g. SuiteSparse ...), and link to it. See this [page](http://eigen.tuxfamily.org/dox/TopicCMakeGuide.html) in the doc.

**Comment by PatWie** [comment]
I fixed CMakeLists for Eigen 2.9. Eigen 3.3 gave me the following error:
https://travis-ci.org/PatWie/CppNumericalSolvers/jobs/193314352#L455

Patches are welcome.
